File: patterns/java/src/refactoring_guru/patterns/prototype/example/Shape.java
package refactoring_guru.patterns.prototype.example;

abstract class Shape {
    int x;
    int y;
    String color;

    public Shape() {}

    public Shape(Shape target) {
        if (target != null) {
            this.x = target.x;
            this.y = target.y;
            this.color = target.color;
        }
    }

    @Override
    public abstract Shape clone();
}

File: patterns/cs/src/RefactoringGuru/Prototype/PrototypeExample/Example/Shapes/Shape.cs
using System;

namespace RefactoringGuru.PrototypeEample.Example.Shapes
{
    abstract class Shape
    {
        public int x;
        public int y;
        public String color;

        public Shape() { }

        public Shape(Shape target)
        {
            if (target != null)
            {
                this.x = target.x;
                this.y = target.y;
                this.color = target.color;
            }
        }

        public abstract Shape Clone();
        
        public bool Equals(Object object2)
        {
            if (!(object2.GetType() == typeof(Shape)))
            {
                return false;
            }
            Shape shape2 = (Shape) object2;
            return shape2.x == x && shape2.y == y && shape2.color == color;
        }
    }
}
